packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -878,7 +879,12 @@ function useMemoCache(size: number): Array<any> {
 
   let data = memoCache.data[memoCache.index];
   if (data === undefined) {
-    data = memoCache.data[memoCache.index] = new Array(size);
+    data = memoCache.data[memoCache.index] = new Array(size).fill(
+      MEMO_CACHE_UNSET_SENTINEL,
+    );
```

Comment:
I was curious about the performance of `fill` and ran a microbenchmark: https://jsbench.me/7ol98ws520/1

Looks like the fastest way is:

```js
data = memoCache.data[memoCache.index] = new Array(size);
for (let i=0; i<n; ++i) data[i] = MEMO_CACHE_UNSET_SENTINEL;
```

Reply:
good call, i updated to use this implementation